public-ui/src/app/data-browser/views/survey-view/components/survey-question-react.component.tsx

```diff
@@ -0,0 +1,199 @@
+}
+
+interface State {
+    showAnswers: boolean;
+    questionWithResults: any;
+    surveyCountAnalysis: object;
+    versionAnalysis: object;
+}
+
+export class SurveyQuestionReactComponent extends React.Component<Props, State> {
+    versionAnalysis: any[] = [];
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            showAnswers: false,
+            questionWithResults: null,
+            surveyCountAnalysis: null,
+            versionAnalysis: null
+        };
+    }
+
+    showAnswers(e?) {
+        if (e && e.key !== 'Enter') {
+            return;
+        }
+        if (!this.state.questionWithResults) {
+            this.getAnalysis();
+            this.getCountAnalysis();
+            if (this.props.isCopeSurvey) {
+                this.getSurveyVersionAnalysis();
+            }
+            this.setState({
+                showAnswers: !this.state.showAnswers
+            });
+        } else {
+            this.setState({
+                showAnswers: !this.state.showAnswers
+            });
+        }
+    }
+
+    getAnalysis() {
+        api.getSurveyQuestionResults(this.props.surveyConceptId, this.props.question.conceptId, this.props.question.path)
+            .then(
+                results => {
+                    const questionWithResults = {
+                        countAnalysis: results.items.filter(a => a.analysisId === 3110)[0],
+                        genderAnalysis: results.items.filter(a => a.analysisId === 3111)[0],
+                        ageAnalysis: results.items.filter(a => a.analysisId === 3112)[0],
+                        participantCountAnalysis: results.items.filter(a => a.analysisId === 3203)[0],
+                        // this.processResults(q, this.survey.participantCount);
+                    };
+                    questionWithResults.countAnalysis.results.sort((a1, a2) => {
+                        if (a1.countValue > a2.countValue) {
+                            return -1;
+                        }
+                        if (a1.countValue < a2.countValue) {
+                            return 1;
+                        }
+                        return 0;
+                    });
+
+
+                    this.setState({ questionWithResults: questionWithResults });
+                }
+            )
+            .catch(err => {
+                console.log('Error searching: ', err);
+            });
+    }
+    getCountAnalysis() {
+        api.getCountAnalysis(this.props.surveyConceptId.toString(), 'survey').then(
+            results => {
+                console.log(results, 'results');
+                this.setState({ surveyCountAnalysis: results });
+                // if (this.surveyCountAnalysis) {
+                //     localStorage.setItem('surveyCountAnalysis', JSON.stringify(results));
+                // }
+            });
+    }
+
+    getSurveyVersionAnalysis(): any {
+        api.getSurveyVersionCounts(this.props.surveyConceptId)
```

Comment:
This can be fetched from survey-view.component.ts or removed from there may be to avoid multiple fetches.